### PR TITLE
Add CI job to build and run services

### PR DESCRIPTION
**Pull request by flaper87**
This is the first step towards e2e testing. For now we will start by
building, and running, the various services. In the mean time, we are
also working on getting proper e2e testing to run in CI. A few changes
are still needed for the later to happen but, this PR should get us a
step closer and to a better place.

related to #110

Signed-off-by: Flavio Percoco <flavio@reblaze.com>

### CODE REVIEW

File: deploy/deploy-ci.sh [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+eval $(minikube docker-env)
+
+export DATE="$(date --iso=m)"
+export GITTAG="$(git describe --tag --long --dirty)"
+export DOCKER_DIR_HASH="$(git rev-parse --short=12 HEAD:curiefense)"
+export DOCKER_TAG="$GITTAG-$DOCKER_DIR_HASH"
+export BASEDIR="$(dirname "$(readlink -f "$0")")"
+
+export ROOT_DIR=$(git rev-parse --show-toplevel)
```
Comment (xavier-rbz) [inline deploy/deploy-ci.sh:11]: maybe run shellcheck on this? Not a blocker in my opinion, can be done later.

File: deploy/istio-helm/chart/charts/gateways/templates/deployment.yaml [hunk 2]
```diff
@@ -452,10 +456,16 @@ spec:
           - key: gs.json
             path: gs.json
           secretName: gs
+{{- end }}
+{{- if eq $.Values.global.proxy.curiefense_bucket_type "local-bucket" }}
```
Comment (xavier-rbz) [inline deploy/istio-helm/chart/charts/gateways/templates/deployment.yaml:460]: May I suggest adding this new option to comments in both deploy/curiefense-helm/curiefense/values.yaml and deploy/istio-helm/chart/values.yaml?

**Review by xavier-rbz — APPROVED** [review]
looks like a good start!

File: deploy/curiefense-helm/deploy.sh [hunk 3]
```diff
@@ -22,6 +22,18 @@ else
     echo "Deploying version $DOCKER_TAG for all images"
 fi
 
-helm upgrade --install --namespace curiefense --reuse-values --atomic --debug \
+if ! kubectl get namespaces|grep -q curiefense; then
+	kubectl create namespace curiefense
+    echo "curiefense namespace created"
+fi
+
+helm upgrade --install --namespace curiefense --reuse-values --timeout "10m" --wait \
```
Comment (xavier-rbz) [inline deploy/curiefense-helm/deploy.sh:30]: `--timeout "10m"` is refused by Helm v2. `--timeout "600"` is accepted though.

**Comment by flaper87** [comment]
@xavier-rbz thanks for the review. Will address these comments in a follow-up PR so we can get his one going.

**Pull request merged**